lib/utils.js

```diff
@@ -194,3 +195,21 @@ function getRestifyClientOptions(opts) {
 
 	return (newOpts);
 }
+
+/*
+ * Convert the top level key names to lower case.
+ *
+ * @param {Object} with key names in mixed case
+ * @returns Object with all lower case keys
+ */
+function downcaseKeys(o) {
+	var keys = Object.keys(o);
```

Comment:
Maybe add `assert.object(o)` and `assert.arrayOfString(keys)`, or other form of protection against trying to run `.toLowerCase()` when it's not a string key, or when o is not an object.